src/auth/src/credentials/idtoken.rs

```diff
@@ -27,6 +66,207 @@ use std::sync::Arc;
 use std::time::{Duration, SystemTime, UNIX_EPOCH};
 use tokio::time::Instant;
 
+pub mod mds {
```

Comment:
I like (or at least I am fine) with the new module structure, but why a submodule with a single re-exported type instead of moving the code to a `idtoken/mds.rs`  file? Are you doing that later?

Reply:
See https://github.com/googleapis/google-cloud-rust/pull/3716/files#r2499272671

We can do it later, but that will required to make a lot of the structs and mod on `auth::credentials::{cred_type}` `pub(crate)` to use them in the separated`idtoken` mod.